File: GVO Code v3/quad1to2.pas
unit quad1to2;
interface
uses utils, sysutils,windows;
procedure GoQuad1to2;
implementation
uses Main, OMS68SERMC ;
procedure GoQuad1to2;
Var
  CmdStr,CmdStr1, Cmdstr2 : string ;
   Begin {1 to 2}
     MovingRa := True;
     MovingDec := True;
     RaPos := RaTarget - EastHor ;
     if(RaPos > 0.0)Then RaPos := 24.0 - RaPos;
     RaPos := abs(Rapos);
     Rapos := (Rapos * RaFact)*15 ;
     Str(Rapos:14:0,CmdStr);
     CmdStr := TrimLeft(CmdStr);
     Str(TrkRate,CmdStr2);
     CmdStr1 := 'AX ST;';              { Send RA STOP}
       SendString(CommRecord,pchar(CmdStr));
     CmdStr := 'AX  VL'+xvlslew+' MA'+ CmdStr + ' GD ID; ';      {Send RA SLEW}
       SendString(CommRecord,pchar(CmdStr));
     DecPos := 90.0 - Dectarget;
     If(DecTarget < 0 ) Then  DecPos := 90.0 + Abs(DecTarget);
     DecPos := (DecPos * DecFact);
     Str(Decpos:14:0,CmdStr);
     cmdStr := TrimLeft(cmdStr);
     CmdStr := 'AY  VL'+yvlslew+ ' MA'+ CmdStr + ' GD ID; ';          {DEC SLEW}
        SendString(CommRecord,pchar(CmdStr));
{Scope is SLEWING}
   End;  {1 to 2}
end.

File: GVO Code C++ v1/quad1to2.cpp
#include "quad1to2.h"
#include "GlobalValues.h"
#include "CommUtils.h"

#include <cmath>
#include <iomanip>
#include <sstream>
#include <iostream>
#include <thread>
#include <chrono>

bool GoQuad1to2(double raTarget, double decTarget, double eastHor) {
	movingRA = true;
	movingDEC = true;

	RaPos = raTarget - eastHor;
	if (RaPos > 0.0)
		RaPos = 24.0 - RaPos;

	RaPos = std::abs(RaPos);
	RaPos = RaPos * RAFact * 15.0;

	std::ostringstream raStream;
	raStream << std::fixed << std::setprecision(0) << RaPos;
	std::string raSteps = raStream.str();

	std::string raCmd = "AX  VL" + xvlslew + " MA" + raSteps + " GD ID;";

	if (decTarget < 0.0)
		decPos = 90.0 + std::abs(decTarget);
	else
		decPos = 90.0 - decTarget;

	decPos = decPos * DECFACT;

	std::ostringstream decStream;
	decStream << std::fixed << std::setprecision(0) << decPos;
	std::string decSteps = decStream.str();

	std::string decCmd = "AY  VL" + yvlslew + " MA" + decSteps + " GD ID;";

	std::cout << "GoQuad1to2:\n";
	std::cout << "  raTarget=" << raTarget << " decTarget=" << decTarget << "\n";
	std::cout << "  eastHor=" << eastHor << "\n";
	std::cout << "  RaPos=" << RaPos << " decPos=" << decPos << "\n";
	std::cout << "  RA cmd: [" << raCmd << "]\n";
	std::cout << "  DEC cmd: [" << decCmd << "]\n";

	std::cout << "[GoQuad1to2] before AX send\n";
	bool raOk = SendCommand(raCmd);
	std::cout << "[GoQuad1to2] after AX send, raOk=" << raOk << "\n";

	if (!raOk) {
		std::cout << "Failed to send RA slew command\n";
		return false;
	}

	std::cout << "[GoQuad1to2] before AY send\n";
	bool decOk = SendCommand(decCmd);
	std::cout << "[GoQuad1to2] after AY send, decOk=" << decOk << "\n";

	if (!decOk) {
		std::cout << "Failed to send DEC slew command\n";
		return false;
	}

	return true;
}